hspec-discover/src/Run.hs

```diff
@@ -41,16 +41,18 @@ run args_ = do
   name <- getProgName
   case args_ of
     src : _ : dst : args -> case parseConfig name args of
-      Left err -> do
-        hPutStrLn stderr err
-        exitFailure
+      Left err -> exitError err
       Right conf -> do
         when (configNested conf) (hPutStrLn stderr "hspec-discover: WARNING - The `--nested' flag is deprecated and will be removed in a future release!")
         specs <- findSpecs src
         writeFile dst (mkSpecModule src conf specs)
-    _ -> do
-      hPutStrLn stderr (usage name)
-      exitFailure
+    _ -> exitError $ if "-h" `elem` args_ || "--help" `elem` args_
+            then helpMessage name
+            else usage name
```

Comment:
I would get rid of that case and instead always print something like

```
hspec-discover: unrecognized option `--foo'
Try `hspec-discover --help' for more information.
```

if the user specifies an invalid argument.  This is consistent with what `hspec` and most other tools do and gives the user the chance to discover `--help` himself.